lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -49,19 +53,31 @@ def command_output(cmd)
     def unix_uname_s
       return @uname[:s] if @uname.key?(:s)
 
-      @uname[:s] = command_output("uname -s")
+      if ARGV.include? "--sudo"
```

Comment:
This is pretty repetitive - perhaps we could make a method, `command_prefix`, which conditionally returns either `"sudo"` or `""`, and then all these conditionals would just look like 
```
@uname[:s] = command_output("#{command_prefix} uname -s")
```

Reply:
Ok, actually I do see where we have the problem visible in the stack trace:
        27: from lib/train/transports/ssh_connection.rb:53:in `initialize'
        26: from /lib/train/extras/command_wrapper.rb:166:in `load'
at this point we initialize the CommandWrapper in
/lib/train/transports/ssh_connection.rb:
`53      @cmd_wrapper            = CommandWrapper.load(self, @transport_options)`

meanwhile deeper in the stack we find:
/lib/train/transports/ssh_connection.rb:284:in `block in execute_on_channel'

so... this call is run in the scope where we yet not defined @cmd_wrapper because we just try to do, resulting in:
`283        cmd = @cmd_wrapper.run(cmd) unless @cmd_wrapper.nil?`

to not add sudo because we don't have the wrapper yet because we are just loading it.

So in fact there may be a workaround required most probably at this point.

Let me think about that some more time and I may find a proper solution :-)